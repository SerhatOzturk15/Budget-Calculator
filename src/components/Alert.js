import React from 'react'

const Alert = () => {
    <div>alert</div>
}