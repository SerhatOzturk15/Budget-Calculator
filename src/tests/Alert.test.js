import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json'
import Alert from '../components/Alert';

const setup = (props = {}) => {
    const component = shallow(<Alert {...props}/>);
    return component;
}

describe('Alert component without props', () => {
    let component = setup();

    it('component should render', () => {
        const wrapper = component.find('.alert');
        expect(wrapper.length).toBe(1);
    });

    it('snapshot should match', () => {
        const wrapper = component.find('.alert');
        expect(toJson(wrapper)).toMatchSnapshot();
    });

})

describe('Alert component with props', () => {
    const props = {
        type: 'danger',
        text: 'text'
    }
    let component =setup(props);

    it('component should render .alert-danger class', () => {
        const wrapper = component.find('.alert-danger');
        expect(wrapper.length).toBe(1);
    });


    it('component should render amount input', () => {
        const wrapper = component.find('[data-testid="alert-test"]');
        expect(wrapper.length).toBe(1);
    });

})