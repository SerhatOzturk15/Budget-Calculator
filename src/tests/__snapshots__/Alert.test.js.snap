// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component without props snapshot should match 1`] = `
<div
  className="alert alert-undefined"
  data-testid="alert-test"
/>
`;
